app/colorCoder.js

```diff
@@ -1,99 +1,38 @@
-
-const MajorColorNames = [
-    "WHITE", "RED", "BLACK", "YELLOW", "VIOLET"
-];
-const MinorColorNames = [
-	"BLUE", "ORANGE", "GREEN", "BROWN", "SLATE"
-];
-
-function ColorPair(){
-           this.majorColor;
-           this.minorColor;
-}
-
-ColorPair.prototype.toString=function(){
-	return `MajorColor:${this.majorColor},MinorColr:${this.minorColor}`;
-}
-
-function getColorFromPairNumber(pairNumber)
-{
-	let minorSize = MajorColorNames.length;
-	let majorSize = MinorColorNames.length;
-	
-	if (pairNumber < 1 || pairNumber > minorSize * majorSize)
- 	{
- 		throw `Argument PairNumber:${pairNumber} is outside the allowed range` 
-	}
-	let zeroBasedPairNumber = pairNumber - 1;
-    let majorIndex = parseInt (zeroBasedPairNumber / minorSize);
-    let minorIndex = parseInt(zeroBasedPairNumber % minorSize);
-	let  pair = new ColorPair();
-	pair.majorColor = MajorColorNames[majorIndex];
-	pair.minorColor = MinorColorNames[minorIndex];
-	return pair;
+const ColorPair = require('../app/colorPair');
+const { getColorFromPairNumber, getPairNumberFromColor } = require('../app/colorUtils');
+const printManual = require('../app/printManual');
+
+function test() {
+    let pairNumber = 4;
+    let testPair1 = getColorFromPairNumber(pairNumber);
+    console.log(`[In]Pair Number: ${pairNumber}, [Out] Colors: ${testPair1}`);
+    console.assert(testPair1.majorColor === "WHITE");
+    console.assert(testPair1.minorColor === "BROWN");
+
+    pairNumber = 5;
+    testPair1 = getColorFromPairNumber(pairNumber);
+    console.log(`[In]Pair Number: ${pairNumber}, [Out] Colors: ${testPair1}`);
+    console.assert(testPair1.majorColor === "WHITE");
+    console.assert(testPair1.minorColor === "SLATE");
+
+    pairNumber = 23;
+    testPair1 = getColorFromPairNumber(pairNumber);
+    console.log(`[In]Pair Number: ${pairNumber}, [Out] Colors: ${testPair1}`);
+    console.assert(testPair1.majorColor === "RED");
+    console.assert(testPair1.minorColor === "GREEN");
+
+    let testPair2 = new ColorPair("YELLOW", "GREEN");
+    pairNumber = getPairNumberFromColor(testPair2);
+    console.log(`[In]Colors: ${testPair2}, [Out] PairNumber: ${pairNumber}`);
+    console.assert(pairNumber === 18);
+
+    testPair2 = new ColorPair("RED", "BLUE");
+    pairNumber = getPairNumberFromColor(testPair2);
+    console.log(`[In]Colors: ${testPair2}, [Out] PairNumber: ${pairNumber}`);
+    console.assert(pairNumber === 6);
+
+    console.log("Manual:");
+    console.log(printManual());
```

Comment:
how to test printManual?

Reply:
@venu-shastri Updated the test method with single function and followed YAGNI principle